WeakAuras/Prototypes.lua

```diff
@@ -10702,6 +10701,183 @@ Private.event_prototypes = {
     end,
     automaticrequired = true
   },
+  ["Location"] = {
+    type = "unit",
+    events = function(trigger, untrigger)
+      local events = {}
+      if trigger.use_zoneIds ~= nil
+        or trigger.use_zone ~= nil
+        or trigger.use_subzone ~= nil
+        or trigger.use_mapType ~= nil
+      then
+        tinsert(events, "ZONE_CHANGED")
+        tinsert(events, "ZONE_CHANGED_INDOORS")
+        tinsert(events, "ZONE_CHANGED_NEW_AREA")
+        tinsert(events, "VEHICLE_UPDATE")
+      end
+      if trigger.use_indoors ~= nil then
+        tinsert(events, "MINIMAP_UPDATE_ZOOM")
+        tinsert(events, "ZONE_CHANGED_INDOORS")
+      end
+      if trigger.use_instanceDifficulty ~= nil
+         or trigger.use_instance_type ~= nil
+         or trigger.use_instance_size ~= nil
+      then
+        tinsert(events, "PLAYER_DIFFICULTY_CHANGED")
+      end
+      return {
+        ["events"] = events,
+      }
+    end,
+    internal_events = function(trigger, untrigger)
+      local events = {"INSTANCE_LOCATION_CHECK"};
+      if trigger.use_instance_difficulty ~= nil
+         or trigger.use_instance_type ~= nil
+         or trigger.use_instance_size ~= nil
+      then
+        tinsert(events, "WA_DELAYED_PLAYER_ENTERING_WORLD")
+      end
+      return events;
+    end,
+    force_events = "INSTANCE_LOCATION_CHECK",
+    name = WeakAuras.newFeatureString..L["Location"],
+    init = function(trigger)
+      local ret = [=[
+        local recursive = %s
+        local zoneChecker = Private.ExecEnv.ParseZoneCheck(%q, recursive)
```

Comment:
Put that into a preamble for zoneIds.

Putting it here makes the trigger parse the zone check string each time it's trigered, whereas the preamble runs the code just once.

For the recursive thing, imho recursive should always be true for the 'c'-prefix.